src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * Amount generated for the "Soul" skills, which are cooldowns
+ */
+const SOUL_GAUGE_GENERATION_AMOUNT = 50
+
+export class Gauge extends CoreGauge {
+	@dependency private suggestions!: Suggestions
+
+	private soulGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SOUL_MAX_VALUE,
+		graph: {
+			label: 'Soul Gauge',
+			color: SOUL_GAUGE_COLOR.fade(0.25),
+		},
+	}))
+
+	private shroudGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SHROUD_MAX_VALUE,
+		graph: {
+			label: 'Shroud Gauge',
+			color: SHROUD_GAUGE_COLOR.fade(0.25),
+		},
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		// soul gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS)), this.onSoulGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS_CDS)), this.onSoulGenerationCds)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_CONSUMERS)), this.onSoulConsumption)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.PLENTIFUL_HARVEST.id), this.onPlentifulHarvest)
+
+		// shroud gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_GENERATORS)), this.onShroudGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_CONSUMERS)), this.onShroudConsumption)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.ENSHROUD.id), this.onEnshroud)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onSoulGeneration() {
+		this.soulGauge.generate(BASE_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onSoulGenerationCds() {
+		this.soulGauge.generate(SOUL_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onSoulConsumption() {
+		this.soulGauge.spend(BASE_GAUGE_CONSUMPTION_AMOUNT)
+	}
+
+	private onShroudGeneration() {
+		this.shroudGauge.generate(BASE_GAUGE_GENERATION_AMOUNT)
+	}
+
+	private onPlentifulHarvest() {
+		this.shroudGauge.generate(PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT)
+	}
+
+	private onEnshroud() {
+		this.shroudGauge.spend(ENSHROUD_GAUGE_CONSUMPTION)
+	}
+
+	private onShroudConsumption() {
+		this.shroudGauge.spend(BASE_GAUGE_CONSUMPTION_AMOUNT)
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.SOUL_SLICE.icon,
+			content: <Trans id="rpr.gauge.suggestions.soul-gauge-overcap.content">
+				Try to not overcap Soul gauge since it's required for shroud generation for your highest damaging skills
+				via <DataLink action="ENSHROUD"/>.
+				Pay special attention when using <DataLink action="SOUL_SLICE"/> or <DataLink action="SOUL_SCYTHE"/> as
+				they increase the gauge by <b>{SOUL_GAUGE_GENERATION_AMOUNT}</b>,
+				make sure you have <b>{SOUL_MAX_VALUE - SOUL_GAUGE_GENERATION_AMOUNT}</b> or
+				less in your gauge before using.
```

Comment:
Appreciate the formatting, but I think this could use a little bit less newlines for improved legibility of the code. Some consistency of capitalisation would be useful here too I think, and maybe make it sound a bit more natural. Happy to discuss this one in discord.